testing/esp/esdlcmd/esdlcmd-test.py

```diff
@@ -103,6 +104,108 @@ def validate_results(self):
+        self.run_settings = run_settings
+        self.name = name
+        self.command = command
+        self.esdl_path = (self.run_settings.test_path / 'inputs' / esdl_file)
+        #self.service = service
+        self.xsl_path = xsl_path
+        self.options = options
+        self.output_path = Path(self.run_settings.output_base) / name
+        self.stdout = use_stdout
+        self.expected_err = expected_err
+        if self.stdout:
+            self.args = [
+                str(run_settings.exe_path),
+                self.command,
+                self.esdl_path,
+                '-cde',
+                self.xsl_path,
+            ]
+        else:
+            self.args = [
+                str(run_settings.exe_path),
+                self.command,
+                self.esdl_path,
+                self.output_path,
+                '-cde',
+                self.xsl_path,
+            ]
+
+        if options:
+            self.args.extend(options)
+
+        self.result = None
+
+    def run_test(self):
+        safe_mkdir(self.output_path)
+        logging.debug("Test %s args: %s", self.name, str(self.args))
+        self.result = subprocess.run(self.args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+        if self.expected_err != None and self.expected_err == self.result.stderr:
+            success = True
+        elif self.result.returncode != 0:
+            logging.error('Error running "esdl %s" for test "%s": %s', self.command, self.name, self.result.stderr)
+            success = False
+        else:
+            success = self.validate_results()
+
+        self.run_settings.stats.add_count(success)
+
+    def is_same(self, dir1, dir2):
+        """
+        Compare two directory trees content.
+        Return False if they differ, True is they are the same.
+        """
+        compared = DirectoryCompare(dir1, dir2)
+        if (compared.left_only or compared.right_only or compared.diff_files
+            or compared.funny_files):
+            return False
+        for subdir in compared.common_dirs:
+            if not self.is_same(os.path.join(dir1, subdir), os.path.join(dir2, subdir)):
+                return False
+        return True
+
+    def validate_results(self):
+        """Compare test case results to the known key.
+
+        Return True if the two are identical or False otherwise.
+        """
+        outName = self.output_path
+        key = (self.run_settings.test_path / 'key' / self.name)
+
+        # When output was stdout, write the captured stdout text to a file named 'from-stdout.ecl'
+        # in the output directory. This allows us to use the same method of comparing the key
+        # and result.
+        if self.stdout:
+            if self.result.stdout != None and len(self.result.stdout) > 0:
+                with open((outName / 'from-stdout.ecl'), 'w', encoding='utf-8') as f:
+                    f.write(self.result.stdout)
```

Comment:
It should probably be do an "else return False" here after line 192.  Otherwise, the tests in the next few lines will compare with previously created "from-stdout.ecl" from previous test runs.

Reply:
fixed